includes/oauth/class-google-login.php

```diff
@@ -114,32 +116,40 @@ public static function oauth_callback() {
 		}
 
 		if ( ! wp_verify_nonce( sanitize_text_field( $_GET[ self::AUTH_CALLBACK ] ), self::AUTH_CALLBACK ) ) {
+			self::handle_error( __( 'Nonce verification failed.', 'newspack' ) );
 			wp_die( esc_html__( 'Invalid nonce.', 'newspack' ) );
 			return;
 		}
 
 		if ( ! isset( $_REQUEST['csrf_token'] ) || ! isset( $_REQUEST['access_token'] ) ) {
+			self::handle_error( __( 'CSRF token or access token missing.', 'newspack' ) );
 			wp_die( esc_html__( 'Invalid request', 'newspack' ) );
 			return;
 		}
 
 		$saved_csrf_token = OAuth::retrieve_csrf_token( self::CSRF_TOKEN_NAMESPACE );
 
 		if ( $_REQUEST['csrf_token'] !== $saved_csrf_token ) {
-			Logger::error( 'Failed saving email - CSRF token mismatch.' );
+			self::handle_error( __( 'CSRF token verification failed.', 'newspack' ) );
 			\wp_die( \esc_html__( 'Authentication failed.', 'newspack' ) );
 		}
 
 		$user_email = Google_OAuth::validate_token_and_get_email_address( sanitize_text_field( $_REQUEST['access_token'] ), self::REQUIRED_SCOPES );
 		if ( is_wp_error( $user_email ) ) {
-			Logger::error( 'Failed validating user: ' . $user_email->get_error_message() );
+			/* translators: %s is the error message */
+			self::handle_error( sprintf( __( 'Failed validating user: %s', 'newspack-plugin' ), $user_email->get_error_message() ) );
 			\wp_die( \esc_html__( 'Authentication failed.', 'newspack' ) );
 		}
 
 		Logger::log( 'Got user email from Google: ' . $user_email );
 
 		// Associate the email address with the a unique ID for later retrieval.
-		set_transient( self::EMAIL_TRANSIENT_PREFIX . OAuth::get_unique_id(), $user_email, 20 );
+		$has_set_transient = set_transient( self::EMAIL_TRANSIENT_PREFIX . OAuth::get_unique_id(), $user_email, 60 * 5 );
```

Comment:
Suggestion: It might be a good idea to move the transient time into a constant so it's easier to find and change in the future.